src/main/java/com/linkedin/kmf/services/TopicManagementService.java

```diff
@@ -10,417 +10,57 @@
-        bldr.append(replica).append(",");
-      }
-      bldr.setLength(bldr.length() - 1);
-      bldr.append("]},\n");
-    }
-    bldr.setLength(bldr.length() - 2);
-    bldr.append("]}");
-    return bldr.toString();
-  }
-
-  /**
-   * This runs the preferred replica election for all partitions.
-   */
-  protected void triggerPreferredLeaderElection(ZkUtils zkUtils, List<PartitionInfo> partitionInfoList) {
-    scala.collection.mutable.HashSet<TopicAndPartition> scalaPartitionInfoSet = new scala.collection.mutable.HashSet<>();
-    for (PartitionInfo javaPartitionInfo : partitionInfoList) {
-      scalaPartitionInfoSet.add(new TopicAndPartition(_topic, javaPartitionInfo.partition()));
-    }
-    PreferredReplicaLeaderElectionCommand.writePreferredReplicaElectionData(zkUtils, scalaPartitionInfoSet);
-  }
-
-  /**
-   * @return the state of the topic e.g. all the partitions and replicas, if the topic does not exist then this will
-   * return null.
-   */
-  private TopicState topicState() {
-    List<PartitionInfo> partitionInfoList = partitionInfo();
-    if (partitionInfoList == null) {
-      return null;
-    }
-    Collection<Broker> brokers = scala.collection.JavaConversions.asJavaCollection(_zkUtils.getAllBrokersInCluster());
-    return topicState(partitionInfoList, brokers, _partitionToBrokerRatioThreshold);
+    Map<String, Object> serviceProps = createMultiClusterTopicManagementServiceProps(props, serviceName);
+    _multiClusterTopicManagementService = new MultiClusterTopicManagementService(serviceProps, serviceName);
   }
 
-  /**
-   *
-   * @return  This will return null if the topic does not exist.
-   */
-  private List<PartitionInfo> partitionInfo() {
-    scala.collection.mutable.ArrayBuffer<String> topicList = new scala.collection.mutable.ArrayBuffer<>();
-    topicList.$plus$eq(_topic);
-    scala.collection.Map<Object, scala.collection.Seq<Object>> partitionAssignments =
-        _zkUtils.getPartitionAssignmentForTopics(topicList).apply(_topic);
-    if (partitionAssignments.isEmpty()) {
-      return null;
-    }
-    List<PartitionInfo> partitionInfoList = new ArrayList<>();
-    scala.collection.Iterator<scala.Tuple2<Object, scala.collection.Seq<Object>>> it = partitionAssignments.iterator();
-    while (it.hasNext()) {
-      scala.Tuple2<Object, scala.collection.Seq<Object>> scalaTuple = it.next();
-      Integer partition = (Integer) scalaTuple._1();
-      scala.Option<Object> leaderOption = _zkUtils.getLeaderForPartition(_topic, partition);
-      Node leader = leaderOption.isEmpty() ?  null : new Node((Integer) leaderOption.get(), "", -1);
-      Node[] replicas = new Node[scalaTuple._2().size()];
-      for (int i = 0; i < replicas.length; i++) {
-        Integer brokerId = (Integer) scalaTuple._2().apply(i);
-        replicas[i] = new Node(brokerId, "", -1);
+  private Map<String, Object> createMultiClusterTopicManagementServiceProps(Map<String, Object> props, String serviceName) {
```

Comment:
It's hard to know if this method is correct. Having some documentation, especially about the output, would be helpful.